components/Login/LoginModal.js

```diff
@@ -93,6 +144,124 @@ class LoginModal extends React.Component {
         });
     }
 
+    socialLogin(e, provider) {
+        e.preventDefault();
+        console.log('Hit on social login icon', provider);
+        this.provider = provider;
+
+        $('.ui.login.modal').modal('toggle');
+
+        //prepare localStorage
+        localStorage.setItem(MODI, 'login');
+        localStorage.setItem(NAME, '');
+
+        //observe storage
+        $(window).off('storage').on('storage', this.handleStorageEvent.bind(this));
+
+        //show hint before open tab
+        const Provider = this.getProviderName();
+        swal({
+            title: 'Information',
+            text: 'A new window of your browser will be opened where you could do a sign in on ' + Provider + '. Please do so.',
+            type: 'info',
+            confirmButtonText: 'Confirm',
+            confirmButtonClass: 'positive ui button',
+            buttonsStyling: false,
+            showCloseButton: false,
+            showCancelButton: false,
+            allowEscapeKey: false
+        })
+        .then(() => {
+            //create new tab
+            let url = 'http://authorizationservice.manfredfris.ch:3000/connect/' + provider;
+
+            let width = screen.width*0.75, height = screen.height*0.75;
+            if (width < 600)
+                width = screen.width;
+            if (height < 500)
+                height = screen.height;
+            let left = screen.width/2-width/2, topSpace = screen.height/2-height/2;
+
+            let win = window.open(url, '_blank', 'width='+width+',height='+height+',left='+left+',top='+topSpace+',toolbar=No,location=No,scrollbars=no,status=No,resizable=no,fullscreen=No');
+            win.focus();
+
+            return true;
+        })
+        .catch();
+    }
+
+    clickedFacebook(e) {
+        this.socialLogin(e, 'facebook');
+    }
+
+    clickedGoogle(e) {
+        this.socialLogin(e, 'google');
+    }
+
+    clickedGithub(e) {
+        this.socialLogin(e, 'github');
+    }
+
+    handleStorageEvent(e) {
+        console.log('storage event', e.key, localStorage.getItem(e.key));
+        //this is available
+
+        if (e.key !== NAME || localStorage.getItem(MODI) !== 'login')
+            return;
+
+        let data = {};
+        try {
+            data = JSON.parse(localStorage.getItem(e.key));
+        } catch (err) {
+            console.log('Error while parsing data', err);
+            return;
+        }
+        finally {
+            //delete data
+            localStorage.setItem(NAME, '');
+        }
+
+        //add language before send to service
+        let language = navigator.browserLanguage || navigator.language;
```

Comment:
In case this is used elsewhere too, add it as a common function. Seems to me like a common function.